frontend/src/components/atoms/ProjectionRowEntry.tsx

```diff
@@ -0,0 +1,36 @@
+import styled from 'styled-components';
+
+type ITransaction = import('../../declarations/transaction').ITransaction;
+
+interface IProjectionRowEntryProps {
+  readonly type: ITransaction['type'] | undefined;
```

Comment:
In the future, in situations like this, you might want to use an optional type here, specified as this

```tsx
  readonly type?: ITransaction['type'];
```

This tells typescript that it is not necessarily present, but it can be, and if it is, it's a Transaction-type.